semantic-python/src/Language/Python/Tags.hs

```diff
@@ -56,54 +55,55 @@ instance (ToTags l, ToTags r) => ToTagsBy 'Custom (l :+: r) where
   tags' (R1 r) = tags r
 
 instance ToTagsBy 'Custom Py.FunctionDefinition where
-  tags' Py.FunctionDefinition
+  tags' t@Py.FunctionDefinition
     { ann = Loc Range { start } span
     , name = Py.Identifier { bytes = name }
-    , parameters
-    , returnType
     , body = Py.Block { ann = Loc Range { start = end } _, extraChildren }
     } = do
       src <- ask @Source
       let docs = listToMaybe extraChildren >>= docComment src
           sliced = slice src (Range start end)
-      Tags.yield (Tag name Function span (firstLine sliced) docs)
-      tags parameters
-      traverse_ tags returnType
-      traverse_ tags extraChildren
+      Tags.yield (Tag name Function span (Tags.firstLine sliced) docs)
+      gtags t
```

Comment:
We can reuse the generic machinery to compute tags over all of the subterm positions at a stroke.